dbms/src/Common/TargetSpecific.h

```diff
@@ -697,3 +707,85 @@ TIFLASH_TARGET_SPECIFIC_NAMESPACE(
 #undef DECLARE_TYPE
 #undef ENUM_TYPE
 #undef GET_TYPE
+
+// clang-format off
+/** Runtime Dispatch helpers for class members.
+  *
+  * Example of usage:
+  *
+  * class TestClass
+  * {
+  * public:
+  *     TIFLASH_MULTITARGET_FUNCTION(
+  *     TIFLASH_MULTITARGET_FUNCTION_HEADER(int), testFunctionImpl, TIFLASH_MULTITARGET_FUNCTION_BODY((int value)
+  *     {
+  *          return value;
+  *     })
+  *     )
+  *
+  *     void testFunction(int value) {
+  *         if (TargetSpecific::AVX512Checker::runtimeSupport())
+  *         {
+  *             testFunctionImplAVX512(value);
+  *         }
+  *         else if (TargetSpecific::AVXChecker::runtimeSupport())
+  *         {
+  *             testFunctionImplAVX(value);
+  *         }
+  *         else if (TargetSpecific::SSE4Checker::runtimeSupport())
+  *         {
+  *             testFunctionImplSSE4(value);
+  *         }
+  *         else
+  *         {
+  *             testFunctionImpl(value);
+  *         }
+  *     }
+  *};
+  *
+  */
+
+/// Function header
+#define TIFLASH_MULTITARGET_FUNCTION_HEADER(...) __VA_ARGS__
+
+/// Function body
+#define TIFLASH_MULTITARGET_FUNCTION_BODY(...) __VA_ARGS__
+
+#if defined(__GNUC__) && defined(__x86_64__)
+
+/// NOLINTNEXTLINE
+#define TIFLASH_MULTITARGET_FUNCTION(FUNCTION_HEADER, name, FUNCTION_BODY) \
+    FUNCTION_HEADER \
```

Comment:
may be refine the format of the #define？

Reply:
how?